query-sql/src/lib.rs

```diff
@@ -290,19 +322,15 @@ impl QueryFragment for SelectQuery {
     fn push_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         out.push_sql("SELECT ");
         self.projection.push_sql(out)?;
-
-        out.push_sql(" FROM ");
         self.from.push_sql(out)?;
 
-        if self.constraints.is_empty() {
-            return Ok(());
+        if !self.constraints.is_empty() {
```

Comment:
It's theoretically possible: `SELECT 1 WHERE 1 < 2` is valid SQL. I'm not sure what you mean by 'simplify', though — can you elaborate?